analyzers/tests/SonarAnalyzer.TestFramework.Test/Common/SourceGeneratorProviderTest.cs

```diff
@@ -22,23 +22,35 @@
 
 namespace SonarAnalyzer.TestFramework.Test.Common;
 
+#if NET
+
 [TestClass]
 public class SourceGeneratorProviderTest
 {
+    private static AnalyzerFileReference RazorSourceGenerator =>
+        SourceGeneratorProvider.SourceGenerators.Single(x => x.FullPath.EndsWith("Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll"));
+
     [TestMethod]
     public void SourceGenerators_ContainsRazorSourceGenerator() =>
         SourceGeneratorProvider.SourceGenerators.Should().ContainSingle().Which.FullPath.Should().EndWith("Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll");
 
     [TestMethod]
     public void RazorSourceGenerator_HasCorrectPath() =>
         RazorSourceGenerator.FullPath
-            .Should()
-            .Be(Path.Combine(Path.GetDirectoryName(typeof(SourceGeneratorProvider).Assembly.Location), "Dependencies", "Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll"));
+            .Should().Be(Path.Combine(SourceGeneratorProvider.LatestSdkFolder(), "Sdks", "Microsoft.NET.Sdk.Razor", "source-generators", "Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll"));
 
     [TestMethod]
     public void RazorSourceGenerator_LoadsCorrectAssembly() =>
         RazorSourceGenerator.GetAssembly().GetName().Name.Should().Be("Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators");
 
-    private static AnalyzerFileReference RazorSourceGenerator =>
-        SourceGeneratorProvider.SourceGenerators.Single(x => x.FullPath.EndsWith("Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll"));
+    [TestMethod]
+    public void LatestSdkFolder_ReturnsCorrectPath()
```

Comment:
This UT is a bit silly, we are testing by simulating the logic on the implementation side. Any ideas are welcomed